File: 2490.回环句.py
#
# @lc app=leetcode.cn id=2490 lang=python3
#
# [2490] 回环句
#

# @lc code=start
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        return [words:=sentence.split(), all(words[i][0] == words[i - 1][-1] for i in range(len(words)))][1]
# @lc code=end

print(Solution().isCircularSentence("Leetcode is cool"))

File: 2490.回环句.cpp
/*
 * @lc app=leetcode.cn id=2490 lang=cpp
 *
 * [2490] 回环句
 */

// @lc code=start
#include<string>
using namespace std;
class Solution {
public:
    bool isCircularSentence(string sentence) {
        if(sentence.front() != sentence.back())return false;
        for(int i=1; i<sentence.length(); ++i){
            if(sentence[i] == ' '){
                if(sentence[i - 1] != sentence[i + 1])return false;
            }
        }
        return true;
    }
};
// @lc code=end

